cedar-policy-core/src/entities.rs

```diff
@@ -48,6 +52,9 @@ pub struct Entities {
     #[serde_as(as = "Vec<(_, _)>")]
     entities: HashMap<EntityUID, Entity>,
 
+    #[serde(skip)]
+    evaluated_entities: Option<EvaluatedEntities>,
```

Comment:
With this change, when we do have the evaluated entities (ie when that field is `Some`), the `Entities` object carries both the evaluated and unevaluated entities.  Would it make more sense to have the `entities` field be an enum, either unevaluated entities (a variant that carries `HashMap<EntityUID, Entity>`) or evaluated entities (a variant that carries `EvaluatedEntities`, or perhaps `HashMap<EntityUID, HashMap<SmolStr, PartialValue>>` directly)?

In addition to being "cleaner" and maybe having the memory footprint of the `Entities` be smaller, this might also prevent us from shooting ourselves in the foot, in that no other code could accidentally look up the unevaluated definitions and evaluate them again, if the evaluated ones are already available.

Reply:
The problem is serialization. You can't throw out the restricted expressions or else you can't serialize anymore

Reply:
I think we don't ever serialize these except for DRT.  Unfortunate that we have to accept performance/memory overhead in the hot path for the sake of DRT, but I'll allow it.  (Or am I wrong?)

Reply:
Same comment as @cdisselkoen - it would be cleaner to use an enum to represent entities that are either evaluated or not evaluated. Make sure to file an issue about this before you close the PR so we don't forget